src/logger/loggerApi.c

```diff
@@ -267,6 +269,29 @@ int api_getCapabilities(struct Serial *serial, const jsmntok_t *json)
         return API_SUCCESS_NO_RETURN;
 }
 
+static void get_wifi_status(struct Serial* serial, const bool more)
+{
+        const struct wifi_ap_cfg* ap_cfg = esp8266_drv_get_ap_config();
+        const struct wifi_client_cfg* clt_cfg = esp8266_drv_get_client_config();
+
+        const bool ap_active = ap_cfg && ap_cfg->active;
+        const bool client_active = clt_cfg && clt_cfg->active;
+        const bool client_connected = esp8266_drv_client_connected();
+
+        json_objStartString(serial, "wifi");
+
+        json_objStartString(serial, "ap");
+        json_bool(serial, "active", ap_active, false);
```

Comment:
Let me see what I can do there.

On Tue, Jul 5, 2016, 18:18 Ryan Doherty <notifications@github.com> wrote:

> In src/logger/loggerApi.c
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/726#discussion_r69659563>
> :
>
> > @@ -267,6 +269,29 @@ int api_getCapabilities(struct Serial *serial, const jsmntok_t *json)
> >          return API_SUCCESS_NO_RETURN;
> >  }
> >
> > +static void get_wifi_status(struct Serial* serial, const bool more)
> > +{
> > +        const struct wifi_ap_cfg* ap_cfg = esp8266_drv_get_ap_config();
> > +        const struct wifi_client_cfg* clt_cfg = esp8266_drv_get_client_config();
> > +
> > +        const bool ap_active = ap_cfg && ap_cfg->active;
> > +        const bool client_active = clt_cfg && clt_cfg->active;
> > +        const bool client_connected = esp8266_drv_client_connected();
> > +
> > +        json_objStartString(serial, "wifi");
> > +
> > +        json_objStartString(serial, "ap");
> > +        json_bool(serial, "active", ap_active, false);
>
> Number of clients connected to the AP would be handy to have, if simply
> for debugging purposes. Also could be useful if someone is seeing weird
> activity with their RCP, would be helpful to know if there was another
> device connected to it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/726/files/b8295b1ee49d9e888201243f72f527558d4e20e5#r69659563>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAIoz_xNeTrMpFSjPMBks43TLY526YWGks5qSvRRgaJpZM4JFpXY>
> .
>